### PR TITLE
feat: Add support for target type in anchor tag xhtml & bbcode

### PR DESCRIPTION
**Pull request by jtoloui-ibm**
# Add support for target type in anchor tag xhtml

## To support target type in xhtml & bbcode links
- Included select option: _blank, _self, _top, _parent

### CODE REVIEW

File: src/formats/bbcode.js [hunk 1]
```diff
@@ -633,13 +640,27 @@
...
 				attrs.defaultattr =
 					escapeEntities(attrs.defaultattr, true) || content;
+				if (target !== undef) {
+					attribs += ' target="' + escapeEntities(target, true) + '"';
+				}
 
-				return '<a href="' + escapeUriScheme(attrs.defaultattr) + '">' +
+				console.log('<a' + attribs + ' href="'
+					+ escapeUriScheme(attrs.defaultattr) + '">' +
+					content + '</a>');
```
Comment (live627) [inline src/formats/bbcode.js:661]:
```suggestion
```

**Review by samclarke — CHANGES_REQUESTED** [review]
Thanks for the PR!

Could you turn off the auto-formatting? All the white space changes make it very difficult to see what's changed.

This won't be able to be merged until the next breaking as this wouldn't be backwards compatible (not all BBCode parser support targets). If you want to make it into a plugin it could be added now otherwise it will have to wait until the next breaking release.

I'm aiming to get as many bugs fixed as possible before the next breaking release. I'm currently planning to go through all the PRs and sort them out in a few weeks and then release another 3.x version. Hopefully, after that it will only take one more release of bug fixes before a new breaking release.

**Comment by jtoloui-ibm** [comment]
> Thanks for the PR!
> 
> Could you turn off the auto-formatting? All the white space changes make it very difficult to see what's changed.
> 
> This won't be able to be merged until the next breaking as this wouldn't be backwards compatible (not all BBCode parser support targets). If you want to make it into a plugin it could be added now otherwise it will have to wait until the next breaking release.
> 
> I'm aiming to get as many bugs fixed as possible before the next breaking release. I'm currently planning to go through all the PRs and sort them out in a few weeks and then release another 3.x version. Hopefully, after that it will only take one more release of bug fixes before a new breaking release.

Sorry, I'll sort out the formatting and also look into creating a plugin if that would make it easier rather to wait for the next major release. 

One question would be, if I go with the plugin route, would you recommend still trying to support BBCode or recommend just creating an xhtml link plugin?

**Comment by jtoloui-ibm** [comment]
Hey @samclarke I've recreated this PR as you suggested in a plugin version #844

**Comment by samclarke** [comment]
Closing this in favour of the plugin version #844.

**Pull request closed without merging**